scripts/base/frameworks/telemetry/main.zeek

```diff
@@ -0,0 +1,498 @@
+##! Module for recording and querying metrics. This modules wraps
+##! the lower-level telemetry.bif functions.
+
+module Telemetry;
+
+export {
+	type labels_table: table[string] of string;
+	type labels_vector: vector of string;
+
+	## Type that captures options used to create metrics.
+	type MetricOpts: record {
+		## The prefix (namespace) of the metric.
+		prefix: string;
+
+		## The human-readable name of the metric.
+		name: string;
+
+		## The unit of the metric. Use the pseudo-unit "1" if this is a unit-less metric.
+		unit: string;
+
+		## Documentation for this metric.
+		helptext: string &default="Zeek Script Metric";
+
+		## The labels names (also called dimensions) of the metric. When
+		## instantiating or working with concrete metrics, corresponding
+		## label values have to be provided.
+		labels: vector of string &default=vector();
+
+		## Whether the metric represents something that is accumulating.
+		## Defaults to ``T`` for counters and ``F`` for gauges and
+		## histograms.
+		is_total: bool &optional;
+
+		## When creating a :zeek:see:`Telemetry::HistogramFamily`,
+		## describes the number and bounds of the individual buckets.
+		bounds: vector of double &optional;
+
+		## The same meaning as *bounds*, but as :zeek:type:`count`.
+		## Only set in the return value of
+		## :zeek:see:`Telemetry::collect_histogram_metrics`.
+		## for histograms when the underlying type is ``int64_t``,
+		## otherwise ignored.
+		count_bounds: vector of count &optional;
```

Comment:
I'm wondering if this is worth is worth having  at all (not very strong either way, just wanted to note).